main.py

```diff
@@ -501,89 +513,45 @@ def main(cl_arguments):
-                load_model_state(
-                    model, pre_target_train, args.cuda, skip_task_models=[], strict=strict
-                )
-            else:  # args.transfer_paradigm == "frozen":
-                # Load the current overall best model.
-                layer_path = get_best_checkpoint_path(args.run_dir, "target_train", task.name)
-                assert layer_path, "No best checkpoint found."
-                load_model_state(
-                    model,
-                    layer_path,
-                    args.cuda,
-                    strict=strict,
-                    skip_task_models=task_names_to_avoid_loading,
-                )
-
     if args.do_full_eval:
-        # Evaluate
         log.info("Evaluating...")
         splits_to_write = evaluate.parse_write_preds_arg(args.write_preds)
-        if args.transfer_paradigm == "finetune":
-            if args.do_target_task_training:
-                for task in target_tasks:
-                    task_to_use = model._get_task_params(task.name).get("use_classifier", task.name)
-                    if task.name != task_to_use:
-                        task_model_to_load = task_to_use
-                    else:
-                        task_model_to_load = task.name
-
-                    # Special checkpointing logic here since we train the sentence encoder
-                    # and have a best set of sent encoder model weights per task.
-                    finetune_path = os.path.join(
-                        args.run_dir, "model_state_%s_best.th" % task_model_to_load
-                    )
-                    if os.path.exists(finetune_path):
-                        ckpt_path = finetune_path
-                    else:
-                        # Find the task-specific best checkpoint to evaluate on.
-                        ckpt_path = get_best_checkpoint_path(
-                            args.run_dir, "target_train", task.name
-                        )
-                    assert ".best" in ckpt_path
-                    load_model_state(
-                        model, ckpt_path, args.cuda, skip_task_models=[], strict=strict
-                    )
 
-                    evaluate_and_write(args, model, [task], splits_to_write)
-            elif args.do_pretrain:
-                # If args.do_target_task_training = 0 and args.do_pretrain = 1
-                # then evaluate on pretraining checkpoints.
-                for task in pretrain_tasks:
-                    ckpt_path = get_best_checkpoint_path(args.run_dir, "pretrain", task.name)
-                    assert ".best" in ckpt_path
-                    load_model_state(
-                        model, ckpt_path, args.cuda, skip_task_models=[], strict=strict
+        if args.do_target_task_training:
+            for task in target_tasks:
+                # Find the task-specific best checkpoint to evaluate on.
+                if args.load_eval_checkpoint:
```

Comment:
We could fold this into the get_best_checkpoint_path logic by including a 'user_specified' flag. If it's not None or 'none' or '', then assert that it exists and return it before doing anything else.